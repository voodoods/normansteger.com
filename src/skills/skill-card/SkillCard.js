import React, { Component } from 'react'
import StyledSkillCard from './SkillCard.styled'
import PropTypes from 'prop-types'

class SkillCard extends Component {
  render() {
    return (
      <StyledSkillCard className="SkillCard">
        <img src={this.props.logo} alt={this.props.name} />
        <h3>{this.props.name}</h3>
        <p>{this.props.description}</p>
      </StyledSkillCard>
    )
  }
}

SkillCard.propTypes = {
  logo: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
}

export default SkillCard
