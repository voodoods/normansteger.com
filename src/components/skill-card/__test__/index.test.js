import React from 'react'
import { shallow } from 'enzyme'
import SkillCard from '../'

describe('<SkillCard /> component index', () => {
  it('expects component to match snapshot', () => {
    const wrapper = shallow(<SkillCard />)
    expect(wrapper).toMatchSnapshot()
  })
})
