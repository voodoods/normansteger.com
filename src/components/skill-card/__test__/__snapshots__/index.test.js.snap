// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkillCard /> component index expects component to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkillCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "centered": undefined,
      "children": Array [
        <div
          className="skill-logo"
        >
          <img />
        </div>,
        <div
          className="skill-story"
        >
          <h3 />
          <p />
        </div>,
      ],
      "className": "SkillCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img />,
          "className": "skill-logo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3 />,
            <p />,
          ],
          "className": "skill-story",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: ",
          [Function],
          ";
  align-items: ",
          [Function],
          ";
  margin: ",
          [Function],
          ";
  width: ",
          [Function],
          ";
  float: ",
          [Function],
          ";

  @media screen and (max-width: 768px) {
    width: ",
          [Function],
          ";
    height: auto;
    margin: ",
          [Function],
          ";
  }

  h3 {
    font-size: 1.4em;
    text-align: ",
          [Function],
          ";
  }
  p {
    font-weight: 300;
    text-align: ",
          [Function],
          ";
  }

  .skill-logo,
  .skill-story {
    display: ",
          [Function],
          ";
  }

  .skill-logo {
    display: flex;
    justify-content: center;
    align-items: center;
    width: ",
          [Function],
          ";
    height: ",
          [Function],
          ";
    margin-right: ",
          [Function],
          ";
    margin-bottom: ",
          [Function],
          ";
    text-align: ",
          [Function],
          ";

    img {
      width: ",
          [Function],
          ";
    }
  }

  .skill-story {
    h3 {
      margin-top: 0;
    }

    width: ",
          [Function],
          ";
    height: ",
          [Function],
          ";
    float: ",
          [Function],
          ";

    @media screen and (max-width: 1024px) {
      width: 100%;
    }
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "centered": undefined,
        "children": Array [
          <div
            className="skill-logo"
          >
            <img />
          </div>,
          <div
            className="skill-story"
          >
            <h3 />
            <p />
          </div>,
        ],
        "className": "SkillCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img />,
            "className": "skill-logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3 />,
              <p />,
            ],
            "className": "skill-story",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  display: ",
            [Function],
            ";
  align-items: ",
            [Function],
            ";
  margin: ",
            [Function],
            ";
  width: ",
            [Function],
            ";
  float: ",
            [Function],
            ";

  @media screen and (max-width: 768px) {
    width: ",
            [Function],
            ";
    height: auto;
    margin: ",
            [Function],
            ";
  }

  h3 {
    font-size: 1.4em;
    text-align: ",
            [Function],
            ";
  }
  p {
    font-weight: 300;
    text-align: ",
            [Function],
            ";
  }

  .skill-logo,
  .skill-story {
    display: ",
            [Function],
            ";
  }

  .skill-logo {
    display: flex;
    justify-content: center;
    align-items: center;
    width: ",
            [Function],
            ";
    height: ",
            [Function],
            ";
    margin-right: ",
            [Function],
            ";
    margin-bottom: ",
            [Function],
            ";
    text-align: ",
            [Function],
            ";

    img {
      width: ",
            [Function],
            ";
    }
  }

  .skill-story {
    h3 {
      margin-top: 0;
    }

    width: ",
            [Function],
            ";
    height: ",
            [Function],
            ";
    float: ",
            [Function],
            ";

    @media screen and (max-width: 1024px) {
      width: 100%;
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
